File: src/OK.ShortLink.Common/Models/LinkModel.cs
using System;

namespace OK.ShortLink.Common.Models
{
    public class LinkModel
    {
        public int Id { get; set; }

        public int UserId { get; set; }

        public UserModel User { get; set; }

        public string Name { get; set; }

        public string Description { get; set; }
        
        public string ShortUrl { get; set; }
        
        public string OriginalUrl { get; set; }

        public bool IsActive { get; set; }

        public DateTime CreatedDate { get; set; }

        public DateTime UpdatedDate { get; set; }
    }
}

File: src/OK.ShortLink.Web/src/app/_models/link.model.ts
import { User } from './user.model';

export class Link {
  constructor(
    public id: number,
    public userId: number,
    public user: User,
    public name: string,
    public description: string,
    public shortUrl: string,
    public originalUrl: string,
    public isActive: boolean,
    public createdDate: Date,
    public updatedDate: Date
  ) {}
}
